### DISCUSSION TITLE
gpspassthrough OK, but no GPS in iNav

### DISCUSSION THREAD

**Discussion by rechtem** [iNavFlight/inav#7960] [Q&A]
Hi,
On the Ports tab, I set the GPS serial port. On the Configuration tab, I ticked GPS checkbox. On the GPS tab, I get "Total messages" count to zero and of course no GPS fix:
![Screenshot from 2022-04-12 18-51-45](https://user-images.githubusercontent.com/72457696/163016764-0fed1df1-27f4-4a2d-9be6-028f60d87276.png)

![Screenshot from 2022-04-12 19-06-16](https://user-images.githubusercontent.com/72457696/163016777-88165b70-32d7-45d3-9a78-2201bdc7e3b0.png)

![Screenshot from 2022-04-12 19-06-26](https://user-images.githubusercontent.com/72457696/163016788-2cfb7586-f7a1-46dd-afaa-d3dcb8ec2865.png)

If on the CLI tab, I issue gpspassthrough, I can monitor the communication with the u-center:

![Screenshot from 2022-04-12 19-10-47](https://user-images.githubusercontent.com/72457696/163017132-f20caf18-ca30-41ce-aa81-7a4e66adb49e.png)

What could be wrong ?
Thanks

**Comment by OptimusTi** [comment]
Show a picture of your connection please (FC & GPS)

**Comment by P-I-Engineer** [comment]
doesn't look like you're connected to ucenter.

what gps? your baud might be too high.

and yes, a picture would be helpful

**Reply by rechtem** [reply]
The above u-center screenshot was done just after power cycling the GPS indoor. I can confirm that if I go outdoor with a laptop, I will quickly get the GPS satellites and position fix.
Also this same drone was using betaflight before I switched to iNav, and **GPS was working fine**.
The GPS is HGLRC M80 https://www.hglrc.com/products/hglrc-m80-gps-for-fpv-racing-drone.

![20220413_080150](https://user-images.githubusercontent.com/72457696/163111921-9798cf96-e83c-417e-9b9a-c91b57892c27.jpg)

![20220413_080218](https://user-images.githubusercontent.com/72457696/163111935-e6642fec-ee73-4c27-9e17-4b9497fb26f8.jpg)

**Comment by rechtem** [comment]
OK, I discovered the GPS has a faulty RX line. To confirm this, I took it off the drone and plugged it in a FTDI usb serial adapter. One cannot send any UBX command which are never ACK/NACK from the device.

**Reply by P-I-Engineer** [reply]
check for a bent pin in the gps connector

**Reply by mrechte** [reply]
Connector is OK. Also checked with another cable.
I wonder whether UBX protocol IN could have been disabled on the UART (UBX-CFG-PORT) ?
